src/ansible_navigator/configuration_subsystem/utils.py

```diff
@@ -0,0 +1,174 @@
+
+    :param internals: The internal storage for settings information
+    :returns: The settings file entry
+    """
+    source = internals.settings_source
+    if source is C.SEARCH_PATH:
+        is_default = True
+    elif source is C.NONE or source is C.ENVIRONMENT_VARIABLE:
+        is_default = False
+
+    default = (
+        "{CWD}/ansible-navigator.{ext} or {HOME}/.ansible-navigator.{ext}"
+        " where ext is yml, yaml or json"
+    )
+
+    return HumanReadableEntry(
+        choices=[],
+        cli_parameters={"short": "None", "long": "None"},
+        current_settings_file=internals.settings_file_path or "None",
+        current_value=internals.settings_file_path or "None",
+        default=default,
+        is_default=is_default,
+        description="The path to the current settings file",
+        name="current_settings_file",
+        env_var="ANSIBLE_NAVIGATOR_CONFIG",
+        settings_file_sample="Not applicable",
+        source=internals.settings_source.value,
+    )
+
+
+def _standard_entry(
+    current: SettingsEntry,
+    application_name: str,
+    settings_file_path: str,
+) -> HumanReadableEntry:
+    """Generate a dictionary containing the details for one settings entry.
+
+    :param current: A settings entry
+    :param application_name: The applications name
+    :param settings_file_path: The current settings file path
+    :return: The settings file entry
+    """
+    if isinstance(current.cli_parameters, CliParameters):
+        cli_long = current.cli_parameters.long_override or f"--{current.name_dashed}"
+
+        if current.cli_parameters.short:
+            cli_short = current.cli_parameters.short
+        else:
+            cli_short = "No short CLI parameter"
+
+    else:
+        cli_long = "No long CLI parameter"
+        cli_short = "No short CLI parameter"
+
+    if isinstance(current.value.current, C):
+        current_value = current.value.current.value
+    else:
+        current_value = current.value.current
+
+    if isinstance(current.value.default, C):
+        default = current.value.default.value
+    else:
+        default = current.value.default
+
+    env_var = current.environment_variable(application_name.upper())
+
+    is_default = current.value.source in (C.NOT_SET, C.DEFAULT_CFG)
+
+    settings_file_sample = _sample_generator(
+        current.settings_file_path(prefix=application_name.replace("-", "_")),
+    )
+
+    return HumanReadableEntry(
+        choices=list(current.choices),  # May be a tuple e.g. PLUGIN_TYPES
+        cli_parameters={"short": cli_short, "long": cli_long},
+        current_settings_file=settings_file_path or "None",
+        current_value=current_value,
+        default=default,
+        description=current.short_description,
+        env_var=env_var,
+        is_default=is_default,
+        name=current.name,
+        settings_file_sample=settings_file_sample,
+        source=current.value.source.value,
+    )
```

Comment:
Looking at this, I'm pretty sure that they are alternative constructors for `HumanReadableEntry` instances. So they need to be coupled with the class. A common way of implementing such constructors is to use `@classmethods` rather than detached functions. Could you convert them accordingly putting them under `HumanReadableEntry`? Then, the caller could use `HumanReadableEntry.from_settings_entry()` and `HumanReadableEntry.from_settings_file()` (well, maybe the names could be improved if I don't fully see the context here).

Reply:
HumanReadableEntry exists only as a data structure to ensure consistency across the dictionaries returned.  It was not intended to be a "functional"? class.

Reply:
It's not about it being functional or not. It's just that there are a lot of ways of constructing it. And constructors belong to the entity they construct, it's a common pattern. The canonical constructor is `__init__()` but when the object can be constructed from other inputs, these additional constructors are used.
The pattern is
```python
class Thing:
    param1: int
    param2: str

    @classmethod
    def from_other_thing(cls, few, other, things):
        # A series of transformations of `few`, `other`, `things` to infer `param1` and `param2`
        return cls(param1, param2)

Thing(42, 'stuff')  # canonical constructor with the default initializer
Thing.from_other_thing(5, 'asd', {})  # alternative constructor with transformations
```

Here's some explanation of when classmethods are usually expected: https://youtu.be/HTLu2DFOdTg?t=1552

Reply:
Ah! I think this would be good. I misunderstood what you were suggesting.